Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -204,14 +209,16 @@ private class TopSitesDataSource: NSObject, UICollectionViewDataSource {
         if indexPath.item < 6 {
             let cell = collectionView.dequeueReusableCellWithReuseIdentifier(ThumbnailIdentifier, forIndexPath: indexPath) as ThumbnailCell
             cell.textLabel.text = site.title.isEmpty ? site.url : site.title
-            cell.image = nil
-
-            if let url = NSURL(string: site.url) {
-                profile.thumbnails.get(url) { (thumbnail: Thumbnail?) in
-                    if let thumbnail = thumbnail {
-                        cell.image = thumbnail.image
+            let key = SDWebThumbnails.getKey(site.url)
+            if let thumbs = profile.thumbnails as? SDWebThumbnails {
+                cell.imageView.moz_getImageFromCache(NSURL(string: key)!, cache: thumbs.cache, completed: { (img, err, type, url) -> Void in
```

Comment:
The cache key is an internal detail, so it would be nice to not have to expose it to the calling code. Would it make sense to change `moz_getImageFromCache` to take the site URL and do they key lookup internally, instead of having to make a separate method call for the key here?

If not, another question I have is whether `moz_getImageFromCache` should really take a URL. Calling `SDWebThumbnails.getKey()`, then assuming that key will be a URL when we do `NSURL(string: key)!` is more detail than we shouldn't be exposing here -- especially since `moz_getImageFromCache()` just unwraps the URL anyway with `absoluteString`!